### PR TITLE
[DO NOT MERGE]support libfuse2 in Ubuntu:22.04

### PR DESCRIPTION
**Pull request by jasondrogba**
The code provided before was to install libfuse2 in centos 7. Now I have encountered some problems installing libfuse2 in ubuntu22.04 in docker image on different platforms(mac m1 arm64 and linux x86_64) and have fixed them.
This PR is for comparison, do not merge
I think in the end we should create a new branch for ubuntu 22.04

### CODE REVIEW

File: lib/fuse_loop_mt.c [hunk 1]
```diff
@@ -44,6 +45,17 @@ struct fuse_mt {
...
+        __u32   len;
+        __u32   opcode;
+        __u64   unique;
+        __u64   nodeid;
+        __u32   uid;
+        __u32   gid;
+        __u32   pid;
+        __u32   padding;
+};
+
```
Comment (jasondrogba) [inline lib/fuse_loop_mt.c]: struct fuse_in_header copy from `fuse_kernel.h`, which is removed.

File: lib/fuse_loop_mt.c [hunk 2]
```diff
@@ -44,6 +45,17 @@ struct fuse_mt {
 	int error;
 };
 
+struct fuse_in_header {
```
Comment (Kai-Zhang) [inline lib/fuse_loop_mt.c]: do not copy the struct. there should be another way to solve the conflict of the types

File: util/ulockmgr_server.c [hunk 3]
```diff
@@ -124,7 +124,7 @@ static int receive_message(int sock, void *buf, size_t buflen, int *fdp,
 	return res;
 }
 
-static int closefrom(int minfd)
+void closefrom(int minfd)
```
Comment (jasondrogba) [inline util/ulockmgr_server.c:127]: fix error
```
ulockmgr_server.c:127:13: error: static declaration of 'closefrom' follows non-static declaration
  127 | static void closefrom(int minfd)
      |             ^~~~~~~~~
In file included from ulockmgr_server.c:14:
/usr/include/unistd.h:363:13: note: previous declaration of 'closefrom' with type 'void(int)'
  363 | extern void closefrom (int __lowfd) __THROW;
      |             ^~~~~~~~~
```

File: util/ulockmgr_server.c [hunk 4]
```diff
@@ -92,7 +92,7 @@ static int receive_message(int sock, void *buf, size_t buflen, int *fdp,
 
 	cmsg = CMSG_FIRSTHDR(&msg);
 	if (cmsg) {
-		if (!cmsg->cmsg_type == SCM_RIGHTS) {
+		if (cmsg->cmsg_type != SCM_RIGHTS) {
```
Comment (jasondrogba) [inline util/ulockmgr_server.c:95]: fix warning
```
ulockmgr_server.c:95:38: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
   95 |                 if (!cmsg->cmsg_type == SCM_RIGHTS) {
      |                                      ^~
ulockmgr_server.c:95:21: note: add parentheses around left hand side expression to silence this warning
   95 |                 if (!cmsg->cmsg_type == SCM_RIGHTS) {
      |                     ^~~~~~~~~~~~~~~~
      |                     (               )
```

File: include/fuse_kernel.h [hunk 5]
```diff
@@ -88,12 +88,16 @@
 #ifndef _LINUX_FUSE_H
 #define _LINUX_FUSE_H
 
-#include <sys/types.h>
+#ifdef __linux__
+#include <linux/types.h>
+#else
+#include <stdint.h>
```
Comment (jasondrogba) [inline include/fuse_kernel.h:94]: fix error on aarch64/arm64(mac m1)
```
In file included from fuse_loop_mt.c:11:
../include/fuse_kernel.h:93:15: error: conflicting types for 'int64_t'; have 'long long int'
   93 | #define __s64 int64_t
      |               ^~~~~~~
In file included from /usr/include/stdint.h:34,
                 from /usr/lib/gcc/aarch64-linux-gnu/11/include/stdint.h:9,
                 from ../include/fuse_common.h:19,
                 from ../include/fuse_lowlevel.h:26,
                 from fuse_loop_mt.c:9:
/usr/include/aarch64-linux-gnu/bits/stdint-intn.h:27:19: note: previous declaration of 'int64_t' with type 'int64_t' {aka 'long int'}
   27 | typedef __int64_t int64_t;
      |                   ^~~~~~~
In file included from fuse_loop_mt.c:11:
../include/fuse_kernel.h:92:15: error: conflicting types for 'uint64_t'; have 'long long unsigned int'
   92 | #define __u64 uint64_t
      |               ^~~~~~~~
In file included from /usr/include/stdint.h:37,
                 from /usr/lib/gcc/aarch64-linux-gnu/11/include/stdint.h:9,
                 from ../include/fuse_common.h:19,
                 from ../include/fuse_lowlevel.h:26,
                 from fuse_loop_mt.c:9:
/usr/include/aarch64-linux-gnu/bits/stdint-uintn.h:27:20: note: previous declaration of 'uint64_t' with type 'uint64_t' {aka 'long unsigned int'}
   27 | typedef __uint64_t uint64_t;
      |                    ^~~~~~~~
```

**Pull request closed without merging**